Client/mods/deathmatch/logic/CClientGif.cpp

```diff
@@ -0,0 +1,357 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto
+ *  LICENSE:     See LICENSE in the top level directory
+ *
+ *  Multi Theft Auto is available from https://www.multitheftauto.com/
+ *
+ *****************************************************************************/
+
+#include "StdInc.h"
+#include "CClientGif.h"
+
+#define GIF_ALLOCATE(pointer,size,release) pointer = (uint8_t*)realloc((release) ? 0 : pointer,(release) ? size : 0UL)
+#define BIGE 0
+#define _SWAP(h) ((BIGE) ? ((uint16_t)(h << 8) | (h << 8)) : h)
+
+CClientGif::CClientGif(CClientManager* pManager, ElementID ID, CGifItem* p_GifItem)
+    : ClassInit(this), CClientTexture(pManager, ID, p_GifItem)
+{
+    SetTypeName("gif");
+    m_pManager = pManager;
+    stride = GetRenderItem()->m_uiSizeX * 4;
+    UpdateTick();
+    m_pGifDisplay = std::make_unique<CClientGifDisplay>(m_pManager->GetDisplayManager(), this);
+}
+
+CClientGif::~CClientGif(){
+    Unlink();
+}
+
+void CClientGif::Register(std::vector<unsigned char*>&& frms, std::vector<int>&& dls) {
+    frames = frms;
```

Comment:
In order to use the whole power of move semantics you also have to move arrays here: `frames = std::move(frms)` and `delays = std::move(dls)`. But since dls is just moved, defaultDelays should be `defaultDelays = delays` now;

Reply:
fixed!